OpenRA.Game/Server/Server.cs

```diff
@@ -213,7 +213,16 @@ public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData, Ser
 						var preConn = PreConns.SingleOrDefault(c => c.Socket == s);
 						if (preConn != null)
 						{
+							// Drop the client if they have taken more than 5 seconds to respond to the handshake request
+							// They either have an unusably bad connection or are trying something malicious
```

Comment:
This won't help at all against malicious clients, they can simply open a new connection when needed.

Reply:
Opening a new connection will generate a new `AuthToken` blob.

Reply:
C=Client, M=Malicious Server, S=Real Server

1. C connects to M.
2. M connects to S.
3. S sends AuthToken to M.
4. M forwards AuthToken to C.
5. C sends salted hash to M.
6. M forwards salted hash to S.
7. M disconnects C.
8. M has an authenticated connection to S.

I don't see how the changes address this problem.

Reply:
This change is not trying to address that problem.